lib/pure/ioselects/ioselectors_epoll.nim

```diff
@@ -197,7 +197,7 @@ proc unregister*[T](s: Selector[T], fd: int|SocketHandle) =
            "Descriptor $# is not registered in the selector!" % $fdi)
   if pkey.events != {}:
     when not defined(android):
-      if pkey.events * {Event.Read, Event.Write} != {}:
+      if pkey.events * {Event.Read, Event.Write, Event.User} != {}:
```

Comment:
There is a similar line below for Android, should this change be replicated there? (line 240).

Reply:
Also, does this need to be replicated in kqueue/other selectors?

Reply:
Actually, another thing, can we simplify this condition? Reading it is melting my brain. It's equivalent to `{Event.Read, Event.Write, Event.User} notin pkey.events`, right?

Reply:
> 
> 
> There is a similar line below for Android, should this change be replicated there? (line 240).

Good Catch. I can't test on Android, but I think you are correct.

> Also, does this need to be replicated in kqueue/other selectors?

See my discussions with @cheatfate. The TLDR;
- It looks like it was already correct in kqueue
- poll uses slightly different code and doesn't need the fix
- I have applied and tested the fix to both `epoll` and Windows `select` already in this PR

> Actually, another thing, can we simplify this condition? Reading it is melting my brain. It's equivalent to {Event.Read, Event.Write, Event.User} notin pkey.events, right?

It's actually the inverse. It's equivalent to:
`Event.Read in pkey.events or Event.Write in pkey.events or Event.User in pkey.events`

I don't have a preference. The set operation is more concise, (it's very *clever* lol) but I could see how it can be confusing. I went with the set operation because that seems to be how it was done originally. I want to be consistent with the style.

Reply:
@dom96 It would be if `notin` would take two sets, right now its more equivalent to notin and notin and notin etc. . This pattern is used all over the place (Araq seems to love it) :D

Reply:
What I would prefer is:

```
{Event.Read, Event.Write, Event.User} in pkey.events
```

But I guess that's not possible in Nim. Perhaps another PR could make it possible and refactor this condition? :)

Reply:
Actually, that could mean the same thing with an `and` in between :/

Honestly, I would prefer the long variant with the many `or`s.

Reply:
The condition is fine the way it is, learn some basic math.

Reply:
In my latest commit I changed this to be the `or` variant.
Do I need to change it back now?
While I understand the importance of basic set theory, is that really a blocker for this PR?

Reply:
> The condition is fine the way it is, learn some basic math.

Don't be so condescending. I can make a complex boolean condition and tell you to learn logic, that doesn't mean it's not shitty code that can be written more clearly.

Reply:
It's not "shitty code", it's idiomatic Nim to use these set expressions.